nixos/modules/services/matrix/synapse.nix

```diff
@@ -755,8 +755,8 @@ in {
         Group = "matrix-synapse";
         WorkingDirectory = cfg.dataDir;
         ExecStartPre = [ ("+" + (pkgs.writeShellScript "matrix-synapse-fix-permissions" ''
-          chown matrix-synapse:matrix-synapse ${cfg.dataDir}/homeserver.signing.key
-          chmod 0600 ${cfg.dataDir}/homeserver.signing.key
+          chown matrix-synapse:matrix-synapse ${cfg.settings.signing_key_path}
+          chmod 0600 ${cfg.settings.signing_key_path}
```

Comment:
I think 0400 is enough for a signing key. (cc @mweinelt as its author)

Reply:
I don't feel strongly about disallowing write access to the signing key for the homeserver user itself. It probably is sufficient to make it read-only, once it does indeed exist.

Reply:
Can we leave the write bit as a separate issue?

In my environment the file does not exist at all (it's a symlink to ${CREDENTIALS_DIRECTORY}), so as far as I am concerned, the write bit can be anything.